paasta_tools/cli/utils.py

```diff
@@ -507,12 +508,14 @@ def run_paasta_serviceinit(subcommand, master, service, instances, cluster, stre
     command_without_empty_strings = [part for part in command_parts if part != '']
     command = ' '.join(command_without_empty_strings)
     log.debug("Running Command: %s" % command)
-    _, output = _run(command, timeout=timeout, stream=stream)
+    return_code, output = _run(command, timeout=timeout, stream=stream)
+    if return_code != 0:
```

Comment:
`_run` is, from what I understand, just a general purpose function to run commands. It doesn't by itself raise any Exceptions if the command it ran doesn't exit with return code 0. As a little contrived example if you wanted to call `grep` with `_run`, you wouldn't want it to raise an Exception if grep returned a non 0 code. However, for critical things like running ssh, we should inspect the return code and react accordingly.